### PR TITLE
Org new signup

### PR DESCRIPTION
**Pull request by mtc2013**
Roughly done.  Needs some comments/refactoring but we now have a feature flag that will not show the proposed org link until such time as we deem it appropriate.

### CODE REVIEW

**Comment by johnnymo87** [comment]
Is there any reason we're renaming the table? The way I usually see STI done, you leave the base table named as `class Organisation < ActiveRecord::Base` in `app/models/organisation.rb`. Then you have an `app/models/organisations/` directory for classes such as `class Organisations::ProposedOrganisation < Organisation`. We'd have to come up with a name for the type organisations where have now ... perhaps `ActiveOrganisation`?

Also, I don't think we need to set a default for the type column. I believe it will automagically populate with `Organisations::ProposedOrganisations` when you go to create one with that constant.

**Comment by mtc2013** [comment]
We wanted to keep the organization model the same.

**Comment by tansaku** [comment]
Great work here guys - have taken a pass through - here are my notes:

* after_inactive_sign_up_path_for (in registrations_controller) is getting a bit long - ready for refactor if gets any longer ...
* great that some of the long organisation class is now in base organisation - at least from point of view of making files shorter :-/
* propose new org feature tests could be more imperative and less declarative - not a reason not to deploy - something for me to reflect on
* further creation of cucumber steps reminds me of the need for ongoing refactoring there ... wonder if coverage testing can show us steps that are not being used ...
* why has user spec moved to using a real org instead of a mock?
* branch needs synced to develop ...?

**Comment by mtc2013** [comment]
in re the user factory, I think it was already using FactoryGirl... we just switched it to create sequential emails rather than the same one.

**Comment by mtc2013** [comment]
Do you mean more declarative and less imperative?

**Comment by mtc2013** [comment]
@tansaku I've synched to develop.

File: Gemfile.lock [hunk 1]
```diff
@@ -2,36 +2,36 @@ GEM
   remote: http://rubygems.org/
```
Comment (johnnymo87) [inline Gemfile.lock]: I see changes to this file but not to `Gemfile`. Are these changes accidental?

Comment (mtc2013) [inline Gemfile.lock]: Gemfile.lock probably changed when i bundle updated to get rid of an error.  https://github.com/AgileVentures/LocalSupport/issues/150

File: db/schema.rb [hunk 2]
```diff
@@ -11,15 +11,35 @@
 #
```
Comment (johnnymo87) [inline db/schema.rb]: Any idea why `rake db:migrate` is requiring so much change to the schema? Is it due to upgrading the versions of the `rails` or `pg` gems?

Comment (mtc2013) [inline db/schema.rb]: I wondered that too but don't know.

File: Gemfile.lock (hunk 1 above)
Comment (johnnymo87) [inline Gemfile.lock]: It looks like we have the important dependencies locked down to a minor version ... so I guess this is safe to do.

**Comment by tansaku** [comment]
@mtc2013 yes more declarative, less imperative - sorry :-) 

Regarding the user factory, I wasn't asking about the change to the email sequences.  I was was asking why we changed from ```let (:model) { mock_model("Organisation") }``` to ``` let (:model) { Organisation.new }```.  

If we are using the real Organisation the now the User unit test can fail if the Org code changes, which is not what we want of a unit test right? We want to ensure that it only passes or fails based on code in the User model no?

**Comment by mtc2013** [comment]
@tansaku I've changed the user spec to now use a mock org but it is a bit brittle due to how factorygirl interacts with it.  And I've also synched with upstream.

**Comment by tansaku** [comment]
thanks @mtc2013 I'd like to know more about the 'brittleness' you describe

**Comment by mtc2013** [comment]
Because we pass the mock to factory girl like this:

 subject(:user) { create(:user, superadmin: false, organisation: model) }

We have to provide mocking for model to be called on _read_attribute with an arg id:
 let (:model) { mock_model("Organisation", :_read_attribute => 1)}

This may well change in some future release.

**Comment by mtc2013** [comment]
@tansaku will this be pulled in soonish?

**Pull request merged**